core/include/detray/navigation/navigator.hpp

```diff
@@ -756,8 +761,8 @@ class navigator {
     /// @param cfg the navigation configuration
     template <typename track_t>
     DETRAY_HOST_DEVICE inline bool update_kernel(
-        const track_t &track, state &navigation,
-        const navigation::config &cfg) const {
+        const track_t &track, state &navigation, const navigation::config &cfg,
+        const context_type &ctx = {}) const {
```

Comment:
Let's not make this a default argument. People might forget to pass the context otherwise
```suggestion
        const context_type &ctx) const {
```